File: cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("Change owned: %i\n", coins);
}

int get_cents(void)
{
    int cents;
    do
    {
        cents = get_int("How much is your Change: ");
        //Calculate the Cents that the customers are owed
    }
    while (cents < 1);
    return cents;
}

int calculate_quarters(int cents)
{
    int count = 0; //Calculate the Quarters owed
    while (cents >= 25)
    {
        cents -= 25;
        count ++;
    }
    return count;
}

int calculate_dimes(int cents)
{
    int count = 0; //Calculate the DIMES owed
    while (cents >= 10)
    {
        cents -= 10;
        count ++;
    }
    return count;
}

int calculate_nickels(int cents)
{
    int count = 0; //Calculate the NICKELS owed
    while (cents >= 5)
    {
        cents -= 5;
        count ++;
    }
    return count;
}

int calculate_pennies(int cents)
{
    return cents;
}

File: sentimental-cash/cash.py
def get_cents():
    while True:
        try:
            # Prompt the user for the amount of change and convert it to cents
            cents = float(input("How much is your change: ")) * 100

            # Check if the entered value is non-negative
            if cents >= 0:
                return int(cents)  # Return the valid input as an integer
            else:
                print("Please enter a non-negative value.")
        except ValueError:
            # Handle non-numeric input with a message
            print("Please enter a valid number.")


def calculate_quarters(cents):
    count = 0
    # Calculate the number of quarters
    while cents >= 25:
        cents -= 25
        count += 1
    return count


def calculate_dimes(cents):
    count = 0
    # Calculate the number of dimes
    while cents >= 10:
        cents -= 10
        count += 1
    return count


def calculate_nickels(cents):
    count = 0
    # Calculate the number of nickels
    while cents >= 5:
        cents -= 5
        count += 1
    return count


def calculate_pennies(cents):
    # The remaining cents after calculating quarters, dimes, and nickels are pennies
    return cents


def main():
    cents = get_cents()  # Get the amount of change from the user

    quarters = calculate_quarters(cents)
    cents -= quarters * 25

    dimes = calculate_dimes(cents)
    cents -= dimes * 10

    nickels = calculate_nickels(cents)
    cents -= nickels * 5

    pennies = calculate_pennies(cents)

    coins = quarters + dimes + nickels + pennies

    # Print the total number of coins to give the customer
    print(f"Change owned: {coins}")


if __name__ == "__main__":
    main()
